Support JSP with Embedded Jetty.

tsachev opened the pull request:
With this change it is now possible
to create executable wars with embedded jetty
that use Jsp.

I have also sample for jetty 9.3 to add, but it is blocked by #5287

ghillert commented:
@tsachev Looks like executable jars work also (?).

tsachev commented:
Did not tried that, but would be cool if so.

janbartel commented:
@tsachev those changes look good to me from the jetty perspective. The only quibble I have is that the spring-dependencies contain both org.eclipse.jetty:apache-jsp and org.mortbay.jasper:apache-jsp, where the former already has the latter as a transitive dep. Is this because you want release 8.0.27 of org.mortbay.jasper:apache-jsp instead of 8.0.9? If so, I will see if I can get a jetty-9.2.16 release done so that you can remove the duplicate dependency and then spring-boot should be able to use this pull request.

tsachev commented:
@janbartel Yeah I could not make it work with 8.0.9.M3. That's why I decided to fix this in spring boot dependencies until a version of jetty 9.2.x is released with eclipse/jetty.project#371.

janbartel commented:
@tsachev I've released jetty-9.2.16 so you can update your pull request and jsp should work with jetty in spring-boot.

tsachev commented:
@janbartel Thanks, will do so.

tsachev commented:
@janbartel I have updated the PR.
now with #5287 merged I also added the jetty 9.3 jsp sample.
@wilkinsona do you think that jetty version bump should be done in another PR?

wilkinsona commented:
@tsachev I've done the Jetty upgrade

nyilmaz commented:
@ghillert @tsachev It does not work with jar packaging. Resulting artifact does not contain any jar files. What would be the approach here? Maybe change spring-boot-maven-plugin's packaging process?

wilkinsona commented:
@nyilmaz We know it doesn't work with jar packaging. In the [section on JSP limitations](http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#boot-features-jsp-limitations), the documentation says:

```
With Jetty it should work if you use war packaging, i.e. an executable war will work, and will also be deployable to any standard container.
```

The best approach is not to use JSPs. Failing that, war packaging will work.
